modules/gltf/src/glb-arrow-loader.ts

```diff
@@ -0,0 +1,135 @@
+import {Matrix4} from '@math.gl/core';
+import {Accessor, Node, PlatformIO, Primitive, WebIO} from '@gltf-transform/core';
+import {KHRONOS_EXTENSIONS} from '@gltf-transform/extensions';
+import {unweld, uninstance, dequantize} from '@gltf-transform/functions';
+import * as arrow from 'apache-arrow';
+import type {LoaderWithParser, LoaderOptions} from '@loaders.gl/loader-utils';
+import {ArrowTable, DataType, Field, Schema, SchemaMetadata} from '@loaders.gl/schema';
+import {deserializeArrowField, deserializeArrowType} from '@loaders.gl/schema-utils';
+import {GLBLoader} from './glb-loader';
+
+/** GLB Arrow loader options */
+export type GLBArrowLoaderOptions = LoaderOptions & {
+  io?: PlatformIO;
+};
+
+export type ArrowTableTransformList = [ArrowTable, Matrix4][];
+
+/**
+ * GLB Loader -
+ * GLB is the binary container format for GLTF
+ */
+export const GLBArrowLoader = {
+  ...GLBLoader,
+  dataType: null as unknown as ArrowTableTransformList,
+  batchType: null as never,
+  worker: false,
+  parse,
+  parseSync: undefined
+} as const satisfies LoaderWithParser<ArrowTableTransformList, never, GLBArrowLoaderOptions>;
+
+async function parse(
+  arrayBuffer: ArrayBuffer,
+  options?: GLBArrowLoaderOptions
+): Promise<ArrowTableTransformList> {
+  const io = options?.io || new WebIO().registerExtensions(KHRONOS_EXTENSIONS);
+  const document = await io.readBinary(new Uint8Array(arrayBuffer));
+
+  // Unclear how represent indexed, instanced, or normalized meshes as
+  // ArrowTable. Convert to simpler representations for now.
+  await document.transform(unweld(), uninstance(), dequantize());
```

Comment:
Pre-processing to ensure the vertex buffers can be represented as Arrow tables. Possibly quantized meshes (int8 or int16 vertex attributes) could be represented, but we'd need to put the 'normalized: boolean' option somewhere.

Similarly there could be ways to represent instanced draws with a second Arrow table for the instance transforms, if we want to go that direction.

Reply:
My thoughts about indexed tables was to add an index column as a `List<Uint32>` and put all the indexes in the first row. We'd need to store an offset to the end of the index list in every other row (in case we had such rows).

```ts
const indexes = new Uint32Array([0, 1, 2, 3, 4, 5]);
const nextIndex = indexes.length;
const indexOffsets = new Uint32Array(5).fill(nextIndex);
```

Reply:
We can add metadata on every column (field) has a `metadata: Map<string, string>` and we can even add a `mesharrow: JSON.stringify(...)` key there.